### PR TITLE
factoriolab: init at 3.8.1

### LINKED ISSUE #356733: Package request: factoriolab
## Project description

[FactorioLab](https://factoriolab.github.io) is a tool for calculating resource and factory requirements for factory games like Factorio, Dyson Sphere Program, Satisfactory, Captain of Industry, Techtonica and Final Factory.

## Metadata

* homepage URL: https://factoriolab.github.io
* source URL: https://github.com/factoriolab/factoriolab
* license: mit
* platforms: unix, linux, darwin

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by PatrickDaG**
Closes: #356733 
[FactorioLab](https://factoriolab.github.io/) is a tool for calculating resource and factory requirements for factory games like Factorio, Dyson Sphere Program, Satisfactory, Captain of Industry, Techtonica and Final Factory.
## Things done

Packaged the project. Since it's a static website it doesn't do very much on its own you need some kind of webserver to host it, such as nginx, or for personal deployment `python -m http.server`.  Could write a nixos module for that but not sure that's necessary.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fa/factoriolab/package.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  nix-update-script,
+}:
+buildNpmPackage rec {
+  pname = "factoriolab";
+  version = "3.8.1";
+
+  src = fetchFromGitHub {
+    owner = "factoriolab";
+    repo = "factoriolab";
+    rev = "v${version}";
```
Comment (steeleduncan) [inline pkgs/by-name/fa/factoriolab/package.nix:18]: `refs/tags/v${version}` is safer

**Review by steeleduncan — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by nicoonoclaste — COMMENTED** [review]
Thanks a bunch for packaging factoriolab  :heart: 

I have some minor feedback, but this looks otherwise ready to ship, though I'll try and test it locally first.

File: pkgs/by-name/fa/factoriolab/package.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    pkg-config
+    makeWrapper
+  ];
+
+  npmDepsHash = "sha256-FX9zxq3kIQ5nxEPW57X53novtCFBPV5w9jg882EC2Lo=";
+  postPatch = ''
+    ${lib.getExe jq} '.projects.factoriolab.architect.build.options += { "optimization": {"fonts": false } }' ./angular.json > angular.json.new
+    mv -f angular.json.new angular.json
+  '';
+  dontFixup = true;
```
[the remark this answers is not in the document]
Comment (PatrickDaG) [reply pkgs/by-name/fa/factoriolab/package.nix:32]: will do.

File: pkgs/by-name/fa/factoriolab/package.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    mkdir -p $out/share
+
+    cp -r dist/browser $out/share/factoriolab
+
+    runHook postInstall
+  '';
+  passthru = {
+    updateScript = nix-update-script { };
+  };
```
[the remark this answers is not in the document]
Comment (PatrickDaG) [reply pkgs/by-name/fa/factoriolab/package.nix:45]: true. yeah I copied that part from another package where there were more passthrus and just forgot to simplify it.

File: pkgs/by-name/fa/factoriolab/package.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (PatrickDaG) [reply pkgs/by-name/fa/factoriolab/package.nix]: Turns out it isn't anymore, I just forgot to remove it from a previous version of the package. Good catch.

**Review by nicoonoclaste — APPROVED** [review]
LGTM.  I'm recovering from a long trip home, and I'm not up for merging now as a result (though someone else could!) so I'll try to remember to test & merge tomorrow.  :)

**Pull request merged**